File: jvm/core/src/main/java/se/idkollen/client/PollOptions.java
package se.idkollen.client;

/** Options controlling the behaviour of the high-level {@code waitFor*} polling helpers. */
public record PollOptions(long intervalMs, long timeoutMs) {
    /** Create options with default values: 2 s interval, 300 s timeout. */
    public PollOptions() {
        this(2_000L, 300_000L);
    }
}

File: csharp/src/Idkollen.Client/PollOptions.cs
namespace Idkollen.Client;

public sealed record PollOptions(TimeSpan? Interval = null, TimeSpan? Timeout = null)
{
    public TimeSpan EffectiveInterval => Interval ?? TimeSpan.FromSeconds(2);
    public TimeSpan EffectiveTimeout => Timeout ?? TimeSpan.FromMinutes(5);
}
